server/budgettracker.data/Repositories/BudgetRepository.cs

```diff
@@ -1,16 +1,39 @@
 using budgettracker.common.Models;
+using budgettracker.data.Converters;
+using budgettracker.data.Models;
 using budgettracker.data.Repositories.Interfaces;
-
+using System;
+using System.Data.Common;
 using System.Threading.Tasks;
 
 namespace budgettracker.data.Repositories
 {
     public class BudgetRepository : IBudgetRepository
     {
-        public Task CreateBudget(Budget budget)
+
+        private readonly BudgetTrackerContext _dbContext;
+        private readonly BudgetConverter _budgetConverter;
+
+        public BudgetRepository(BudgetTrackerContext dbContext)
+        {
+            _dbContext = dbContext;
+            _budgetConverter = new BudgetConverter();
+        }
+
+        public async Task CreateBudget(Budget budget)
         {
-            // TODO: Save the created budget only
-            throw new System.NotImplementedException();
+            BudgetModel newBudget = _budgetConverter.ToDataModel(budget);
+
+            using (_dbContext)
+            {
+                await _dbContext.Budgets.AddAsync(newBudget);
+                int recordSaved = await _dbContext.SaveChangesAsync();
+
+                if(recordSaved != 1) 
+                {
+                    throw new Exception("Created " + recordSaved + " budget(s) when only 1 should have been created");
```

Comment:
In order to make this useful, the caller would have to catch Exception... which is bad because that catches literally everything, even stuff we don't want to catch. So instead, make an exception class called RepositoryException that denotes that data access failed. Then the caller can catch that without catching stuff like IO exceptions or system level exceptions.

Reply:
That being said, this here is a good use of "Tell don't ask". 👍

Reply:
Yeah I was thinking about something like that when I deved this.